drf_kit/models.py

```diff
@@ -128,24 +127,33 @@ class OrderedModelMixin(_OrderedModelBase):
     order_field_name = "order"
 
     def save(self, *args, **kwargs):
-        if getattr(self, self.order_field_name) is None:
-            highest_order = (
-                self.get_ordering_queryset().aggregate(Max(self.order_field_name)).get(self.order_field_name + "__max")
-            )
-            new_order = 0 if highest_order is None else highest_order + 1
-            setattr(self, self.order_field_name, new_order)
-        super().save(*args, **kwargs)
+        if kwargs.pop("ignore_order_validation", False):
+            return super().save(*args, **kwargs)
 
-        new_order = getattr(self, self.order_field_name, None)
-        if not new_order:
-            return
+        # How to prevent the order indexes from becoming sparse:
+        # 1. Fetch the list of objects of that grouping, except the current item
+        # 2. Insert the current item in the desired index
+        #    a. if the current item does not have a specified index, add to the end
+        #    b. if it does, add to that index
+        # 3. Traverse the list and update each item whose index has changed
 
-        clash = self.get_ordering_queryset().filter(**{self.order_field_name: new_order}).exclude(pk=self.pk).first()
-        if not clash:
-            return
+        order = getattr(self, self.order_field_name)
 
-        setattr(clash, self.order_field_name, new_order + 1)
-        clash.save()
+        group = list(self.get_ordering_queryset().exclude(id=self.pk))
+
+        if order:  # when updating
```

Comment:
Should be `if order is not None` to handle cases that `order=0` intentionally.

Reply:
It'd be good to have an additional test for this too.
"Move an item from 3 to 0"

Reply:
Good catch!